### PR TITLE
favorite and unfavorite view package

### PR DESCRIPTION
**Pull request by FarhanAliRaza**
fixes https://github.com/djangopackages/djangopackages/issues/1213 https://github.com/djangopackages/djangopackages/issues/1212
[Screencast from 2024-07-25 23-15-27.webm](https://github.com/user-attachments/assets/93dedb99-e0a2-4840-8a6c-9f2c67defb20)

### CODE REVIEW

**Comment by jefftriplett** [comment]
@FarhanAliRaza Looks good. I'll pull and test out messages tomorrow. I saw an article a while back that might have something useful. Otherwise, messages don't need to hold it up. 

Good work!

**Comment by FarhanAliRaza** [comment]
So do we load the Favorite for Package in Package Detail View in this pr or create another issue?

**Comment by jefftriplett** [comment]
> So do we load the Favorite for Package in Package Detail View in this pr or create another issue?

Yes, that's what I would do. I would update the template `package.html` to show that the package is favorited when we render the page. 

It might be a good excuse to look at https://github.com/carltongibson/django-template-partials which I use in a bunch of other projects. I'm not sure if this is a good fit, but I have used it for star'ing images and it was fine.

**Comment by jefftriplett** [comment]
This looks good. I'll check it out Friday locally and do a few hand tests. Good work!

File: templates/package/partials/favorites.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        <h3>
+            Do you want to unfavorite {{package.title}}?
+        </h3>
+        <div class="action ">
+            <button class="btn btn-default" onclick="closeModal()" id="no">
+                No
+            </button>
+            <button class="btn btn-default" id="yes" hx-post="{% url 'remove_favorite' package.id %}" hx-target="#favorite" hx-swap="innerHTML">
+                Yes
+            </button>
```
Comment (jefftriplett) [inline templates/package/partials/favorites.html:28]: I am going to swap the order of these. 

```suggestion
            <button class="btn btn-default" id="yes" hx-post="{% url 'remove_favorite' package.id %}" hx-target="#favorite" hx-swap="innerHTML">
                Yes
            </button>
            <button class="btn btn-default" onclick="closeModal()" id="no">
                No
            </button>
```

**Review by jefftriplett — APPROVED** [review]
@FarhanAliRaza great job! This looks good. I'm going to merge and do a follow up with a feature flipper so we can test it and deploy it when everything else is rady. 

Good work!

**Pull request merged**